salt/tests/unit/modules/files/test_metalk8s_service_configuration.yaml

```diff
@@ -0,0 +1,104 @@
+get_service_config:
+  #. Error: missing configmap_name
+  - configmap_name: null
+    result: "Expected a ConfigMap name but got None"
+    raises: True
+
+  #. Error: default_csc is not of type `dict`
+  - configmap_name: 'my_configmap'
+    default_csc: 'my_default_csc'
+    result: "Expected default CSC for ConfigMap my_configmap but got my_default_csc"
+    raises: True
+
+  # #. Error: unable to retrieve a ConfigMap object
+  - configmap_name: 'my_configmap'
+    object_get: False
+    result: "Failed to read ConfigMap object my_configmap"
+    raises: True
+
+  #. Error: retrieved ConfigMap manifest is empty
+  - configmap_name: 'my_configmap'
+    object_get: True
+    configmap_obj: null
+    result: "Expected ConfigMap object but got None"
+    raises: True
+
+  #. Error: yaml error raised when reading config.yaml
+  - configmap_name: 'my_configmap'
+    configmap_obj:
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: metalk8s-dex-config
+      data:
+        config.yaml: |-
+          {{ my_config.yaml }}
```

Comment:
For testing purpose I will prefer you use some "obvious" invalid YAML instead of some kind of "non-rendered" jinja